merlin/models/tf/transforms/features.py

```diff
@@ -743,17 +743,29 @@ def call(
         target_columns = self._target_column_schemas()
 
         outputs = {}
-        for name, val in inputs.items():
+        for name in inputs:
             if name not in target_columns:
                 outputs[name] = inputs[name]
                 continue
             if isinstance(targets, dict):
-                targets[name] = targets.get(name, inputs[name])
+                _target = targets.get(name, inputs[name])
+                if self.one_hot:
+                    _target = self._to_one_hot(name, _target)
+                targets[name] = _target
             else:
-                targets = inputs[name]
+                _target = inputs[name]
+                if self.one_hot:
+                    _target = self._to_one_hot(name, _target)
+                targets = _target
 
         return outputs, targets
 
+    def _to_one_hot(self, name, target):
+        num_classes = self.schema[name].int_domain.max + 1
```

Comment:
Can we use the `schema_utils.categorical_cardinalities()` method instead? Just to standardize how we access the custom properties of the ColumnSchema (and avoid modifying the code in different places if these properties change in merlin-core)